### DISCUSSION TITLE
ShellBar & setTheme

### DISCUSSION THREAD

**Discussion by Qcyk** [UI5/webcomponents-react#5113] [Q&A] [answered]
Hello, 

I have two questions: 

1. Is it possible to add custom div to ShellBar component? ShellBarItem doesn't have possibility to add custom react element. We can add only icon. 

2. setTheme doesn't cause re-render a children components in ThemeProvider. Method getTheme() is executed only once when component is rendering first time. Is it possible to force components that use getTheme() to re-render (without custom ThemeProvider)?

Regards,
Krzysztof

**Comment by Lukas742** [comment, marked as answer]
Hi @Qcyk 

> Is it possible to add custom div to ShellBar component? ShellBarItem doesn't have possibility to add custom react element. We can add only icon.

Unfortunately this is not supported, only `ShellBarItem`s should be passed as `children`.

> setTheme doesn't cause re-render a children components in ThemeProvider. Method getTheme() is executed only once when component is rendering first time. Is it possible to force components that use getTheme() to re-render (without custom ThemeProvider)?

All children of the `ThemeProvider` should update their style if `setTheme` is called and the theme changes. If this is not the case in your app, then this sounds like a bug to me. [Here](https://codesandbox.io/p/sandbox/billowing-glade-jmj95w?file=%2Fsrc%2FApp.tsx%3A24%2C46) you can find a basic codeSandbox how changing themes should work. 

If you want to register to a theme change, then there is currently no public API for that. If you feel like this should be available then please create a [feature request](https://github.com/SAP/ui5-webcomponents/issues) in the ui5-webcomponents repo, because our colleagues there are responsible for all theming features.
